.github/workflows/pr_verification.yml

```diff
@@ -0,0 +1,355 @@
+          DOCKER_CONFIG=${DOCKER_CONFIG:-$HOME/.docker}
+          mkdir -p $DOCKER_CONFIG/cli-plugins
+          curl -SL https://github.com/docker/compose/releases/download/v2.18.1/docker-compose-linux-x86_64 -o $DOCKER_CONFIG/cli-plugins/docker-compose
+          sudo chmod +x $DOCKER_CONFIG/cli-plugins/docker-compose
+
+      - name: Log in to the registry
+        if: ${{ ! github.event.pull_request.head.repo.fork }}
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Pull candlepin image
+        if: ${{ ! github.event.pull_request.head.repo.fork }}
+        shell: bash
+        run: docker image pull ghcr.io/candlepin/gha_build:pr-${{ github.event.pull_request.number }}
+
+      - name: Set up Docker Buildx
+        if: ${{ github.event.pull_request.head.repo.fork }}
+        uses: docker/setup-buildx-action@v2
+
+      - name: Download artifact
+        if: ${{ github.event.pull_request.head.repo.fork }}
+        uses: actions/download-artifact@v3
+        with:
+          name: gha_build
+          path: /tmp
+
+      - name: Load image
+        if: ${{ github.event.pull_request.head.repo.fork }}
+        run: docker load --input /tmp/gha_build.tar
+
+      - if: matrix.mode == 'standalone'
+        name: Add standalone env variable
+        working-directory: ./.github/containers
+        shell: bash
+        run: |
+          echo "CANDLEPIN_STANDALONE=true" > .env
+          echo "CANDLEPIN_AUTH_CLOUD_ENABLE=false" >> .env
+
+      - if: matrix.mode == 'hosted'
+        name: Add hosted env variables
+        working-directory: ./.github/containers
+        shell: bash
+        run: |
+          echo "CANDLEPIN_STANDALONE=false" > .env
+          echo "MODULE_CONFIG_HOSTED_CONFIGURATION_MODULE=org.candlepin.hostedtest.AdapterOverrideModule" >> .env
+          echo "CANDLEPIN_AUTH_CLOUD_ENABLE=true" >> .env
+
+      - name: Create Candlepin and database containers
+        shell: bash
+        run: |
+          echo "CANDLEPIN_IMAGE=ghcr.io/candlepin/gha_build:pr-${{ github.event.pull_request.number }}" >> ./.github/containers/.env
+          BRIDGE_NETWORK=$(docker network ls --filter=name=github_network_ --format="{{ .Name }}")
+          echo "NETWORK=$BRIDGE_NETWORK" >> ./.github/containers/.env
+          docker compose -f ./.github/containers/${{ matrix.database }}.docker-compose.yml --env-file ./.github/containers/.env up -d --wait
+
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: ${{ env.JAVA_DISTRIBUTION }}
+          java-version: ${{ env.JAVA_VERSION }}
+
+      - name: Run Spec tests
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: spec -Dspec.test.client.host=candlepin
+          
+      - name: Collect docker logs on failure
+        if: failure()
+        uses: jwalton/gh-docker-logs@v2
```

Comment:
Just a note on this, we have another comment on the use of this GH action and we are investigating on how to proceed on this.

https://github.com/candlepin/candlepin/pull/4190/files#r1310527992

Reply:
I took a look at jwalton/gh-docker-logs and it looks good to me to use. We should pin the action to the current SHA to prevent additional security risk!